### PR TITLE
Add Colab notebook

### PR DESCRIPTION
**Pull request by farmaker47**
Add notebook to build TensrFlow Lite Support libraries with Bazel inside Colaboratory

### CODE REVIEW

File: tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb
Comment (xunkai55) [inline tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    %env ANDROID_SDK_API_LEVEL=29
</pre><p>We actually use ANDROID_SDK_API_LEVEL=26 internally. And the build tools version we used is 28.0.0. But probably it doesn't matter very much.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tflite-support/pull/620/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (xunkai55) [inline tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #2.    # Download Bazel
</pre><p>Remove the bazel downloading part?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tflite-support/pull/620/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: I think it doesn't matter. It seems to run also with newer version 29 and 29.0.2.

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: Removed

File: tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb
Comment (lu-wang-g) [inline tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <p>nit: please give a subtitle to this section at the same level of "Install BAZEL with Baselisk".</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tflite-support/pull/620/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (lu-wang-g) [inline tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <p>Can you please instructions for C++ targets as well?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tflite-support/pull/620/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (lu-wang-g) [inline tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    library = '//tensorflow_lite_support/java:tensorflowlite_support.aar' #@param ["//tensorflow_lite_support/java:tensorflowlite_support.aar", "//tensorflow_lite_support/java/src/java/org/tensorflow/lite/task/vision:task-library-vision", "//tensorflow_lite_support/java/src/java/org/tensorflow/lite/task/text:task-library-text", "//tensorflow_lite_support/java/src/java/org/tensorflow/lite/task/audio:task-library-audio","//tensorflow_lite_support/metadata/java:tensorflow-lite-support-metadata-lib"]
</pre><p>Better to make the comment more clear, such as one line per aar:</p><p>Support library: "//tensorflow_lite_support/java:tensorflowlite_support.aar"</p><p>Task Vision library: "//tensorflow_lite_support/java/src/java/org/tensorflow/lite/task/vision:task-library-vision"</p><p>Task Text library: "//tensorflow_lite_support/java/src/java/org/tensorflow/lite/task/text:task-library-text"</p><p>...</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tflite-support/pull/620/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (lu-wang-g) [inline tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    !bazel build -c opt --cxxopt='--std=c++14' \
</pre><p>--cxxopt='--std=c++14' is the default in our bazelrc file.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tflite-support/pull/620/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (lu-wang-g) [inline tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #6.        --host_crosstool_top=@bazel_tools//tools/cpp:toolchain \
</pre><p>This is default in our bazelrc as  well.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tflite-support/pull/620/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: Removed

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: Removed

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: Configured so the user can see only "Support Library" and underneath it takes the value "//tensorflow_lite_support/java:tensorflowlite_support.aar"

Comment (lu-wang-g) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <p>What do you mean by "user can see only "Support Library""? Is "Support Library" an option? I couldn't find it anywhere in this colab.</p>

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: Sorry for the misunderstanding. Check the newest commit.

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: I did not catch this. Can you show me an example and the line of code?

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: I did not catch this. Can you show me an example and the line of code?

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: lu

Comment (lu-wang-g) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <p>I mean adding a subtitle for this section, i.e. "Set up Android environment", which is similar to the subtitle, "Install BAZEL with Baselisk", you added to the section below.</p>

Comment (lu-wang-g) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <p>The C++ targets are:</p><p> </p><p>bazel build -c opt \</p><p>&nbsp;&nbsp;&nbsp;//tensorflow_lite_support/cc/task/vision:image_classifier \ </p><p>   //tensorflow_lite_support/cc/task/vision:image_objector \ </p><p>   //tensorflow_lite_support/cc/task/vision:image_segmenter \ </p><p>   //tensorflow_lite_support/cc/task/vision:image_embedder \ </p><p>   //tensorflow_lite_support/cc/task/text/nlclassifier:nl_classifier \ </p><p>   //tensorflow_lite_support/cc/task/text/nlclassifier:bert_nl_classifier \ </p><p>   //tensorflow_lite_support/cc/task/text/qa:bert_question_answerer \ </p><p>&nbsp;&nbsp;//tensorflow_lite_support/metadata/cc:metadata_extractor \</p><p>&nbsp;</p>

Comment (lu-wang-g) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <p>I see I see. It looks great! Thanks, George! </p>

File: tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb
Comment (lu-wang-g) [inline tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #12.        library = '//tensorflow_lite_support/java/src/java/org/tensorflow/lite/task/text:task-library-tex'
</pre><p> </p><p>Should be task-library-text?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tflite-support/pull/620/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (lu-wang-g) [inline tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.        --config=monolithic \
</pre><p> </p><p>This is not needed as well. It's a flag used for tflite select ops, which is not needed here. A lot commands in our repo has this one, but Xunkai and I found it yesterday that it is not needed. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tflite-support/pull/620/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: OK

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: OK

File: tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb
Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: OK

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: So the displayed name to the users will be "image_classifier" "image_objector"...just like "Support Library" "Task Vision" etc. ??

Comment (lu-wang-g) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: What about "C++ image_classifier". "C++ object_detector", etc?

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: OK

File: tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb
Comment (lu-wang-g) [inline tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <p>nit: rename it to Build TFLite Support targets? Since we also have C++ targets here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tflite-support/pull/620/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (lu-wang-g) [inline tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    !wget -q 'https://dl.google.com/android/repository/tools_r25.2.5-linux.zip'
</pre><p> </p><p> </p><p>Does it work on Windows if we download the SDK and NDK for Windows?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/tflite-support/pull/620/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

**Comment by lu-wang-g** [comment]
Thanks for keep polishing the PR, George! Really appreciate it!

I don't have any comments. I'll approve once Xunkai confirms it.

File: tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb
Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: OK

Comment (farmaker47) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb:1]: I have not tried that on windows.

**Review by xunkai55 — APPROVED** [review]
Looks good to me. Thank you George for polishing it and Lu for the thorough review!

File: tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb
Comment (lu-wang-g) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb]: Is the title changed? It still seems to be "Build .aar files"

Comment (lu-wang-g) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb]: OK. No worries!

Comment (lu-wang-g) [reply tensorflow_lite_support/tools/Build_TensorFlow_Lite_Support_Libraries_with_Bazel_and_Colab.ipynb]: Nevermind. Let me change it after the code is merged. Thanks again for the PR!!

**Pull request merged**
